src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -28,9 +28,37 @@
 #include <controller/CommissionerDiscoveryController.h>
 #include <platform/CHIPDeviceLayer.h>
 #include <setup_payload/AdditionalDataPayloadGenerator.h>
+#include <app/app-platform/ContentAppPlatform.h>
 
 #if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY
 
+namespace {
+
+bool fillRotatingIdBuffer(UDCClientState* client, char[] buffer) {
+    auto rotatingIdLength = client->GetRotatingIdLength();
+    auto hexBufferByteCount = 2 * rotatingIdLength;
+    if (sizeof(buffer) < hexBufferByteCount) {
```

Comment:
This code does not do the right thing (and I would expect it to fail to compile with -Werror).

Reply:
Code removed in latest update.